lib/ast/definition.cpp

```diff
@@ -118,7 +123,31 @@ SubsortMap kore_definition::get_subsorts() const {
     }
   }
 
-  return transitive_closure(subsorts);
+  subsorts_ = transitive_closure(subsorts);
+  return *subsorts_;
+}
+
+SubsortMap kore_definition::get_supersorts() {
+
+  if (supersorts_.has_value()) {
+    return *supersorts_;
+  }
+
+  auto supersorts = SubsortMap{};
+
+  for (auto *axiom : axioms_) {
+    if (axiom->attributes().contains(attribute_set::key::Subsort)) {
+      auto const &att = axiom->attributes().get(attribute_set::key::Subsort);
+      auto const &inner_sort
+          = att->get_constructor()->get_formal_arguments()[0];
+      auto const &outer_sort
+          = att->get_constructor()->get_formal_arguments()[1];
+      supersorts[outer_sort.get()].insert(inner_sort.get());
+    }
+  }
+
+  supersorts_ = transitive_closure(supersorts);
+  return *supersorts_;
```

Comment:
I realise you can actually simplify this a bit more, in fact - down to something like:
```
if (empty) {
  compute sub / supers;
  transitive closure;
}

return *subsorts_;
```

Reply:
To be honest, I would like to do something like:
```
get_subsorts() {
  if (empty) {
    compute sub && compute super
    sub_ = transitive closure sub
    super_ transitive closure super
  }
  
  return *sub_;
}
```

Reply:
I'm not sure that it makes sense to precompute both at the same time; there are a few existing call sites that use the subsorts relation but don't need the supersorts one.